nnvm/python/nnvm/compiler/graph_util.py

```diff
@@ -146,3 +147,39 @@ def gradients(ys, xs, grad_ys=None):
         if isinstance(xs, list) else len(xs.list_output_names())
     ret = [grad_g.symbol[i] for i in range(nx)]
     return ret
+
+def split_last_op(graph):
```

Comment:
I also agree with @zhreshold , lets not limit the split only to the last op node, we can make the function more generic and take input from use like "op_name" and "sequence in the occurrence from the last node", so that the function can be used for formal splitting at any place not only last node...

For example: op_name="flatten", sequence="1" it will split the first occurrence of flatten from the last node. In this way last node split and other node split also can be achieved. 

~/A.T.

Reply:
Agree. But we need a more complete design to cover all cases, especially for networks which have many branches. Simple logics like the last X operators might not work well. A rough idea could be: User select a set of nodes as subgraph inputs and another set of nodes as outputs. Then we do some checks to make sure this is a valid splitting and compile different subgraphs separately.

Reply:
@kevinthesun : you are right, it is little tricky, there are some cases where we can not split the graph as simple, there are many cases needs to be handled, lets create a prototype and discuss on top of that, i am not sure whether in your PR needs to be handled, but i brought the point as it is a very important feature. Thanks! 
@tqchen : please provide your opinion on this, Thanks!

Reply:
My suggestion is to name it more general, but for now only support split on last op (raise exception otherwise). Once you create an API, you just lost chance to remove or rename it.